google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -443,11 +444,23 @@ public boolean deleteLog(String log) {
     return get(deleteLogAsync(log));
   }
 
+  public boolean deleteLog(String log, LogDestinationName destination) {
+    return get(deleteLogAsyncImpl(log, destination));
+  }
+
   public ApiFuture<Boolean> deleteLogAsync(String log) {
-    DeleteLogRequest request =
-        DeleteLogRequest.newBuilder()
-            .setLogName(LogName.ofProjectLogName(getOptions().getProjectId(), log).toString())
-            .build();
+    return deleteLogAsyncImpl(log, null);
+  }
+
+  public ApiFuture<Boolean> deleteLogAsync(String log, LogDestinationName destination) {
+    return deleteLogAsyncImpl(log, destination);
+  }
+
+  private ApiFuture<Boolean> deleteLogAsyncImpl(String log, LogDestinationName destination) {
+    LogName name =
+        Preconditions.checkNotNull(getLogName(getOptions().getProjectId(), log, destination));
```

Comment:
can getOptions() or getOptions().getProjectId() also return null?

Reply:
`getOptions()` might return null since `LoggingImpl` constructor accepts null. Thats said is it safe to assume that `getOptions()` is not null at this point since options are accessed already in `LoggingImpl` constructor. If `getProjectId()` is null it will throw an exception during `LogName` building. I will change a code to give meaningful message for caller since for `DeleteLogRequest` the `LogName` construction is mandatory